java/source/io.rst

```diff
@@ -0,0 +1,578 @@
+.. _arrow-io:
+
+========================
+Reading and writing data
+========================
+
+Recipes related to reading and writing data from disk using Apache Arrow.
+
+Arrow defines two types of binary formats for serializing record batches `IPC <https://arrow.apache.org/docs/java/ipc.html>`_: Streaming format / File or Random Access format
+
+Arrow vectors can be serialized to disk as the Arrow IPC format. Such files can be directly memory-mapped when read.
```

Comment:
I think we can consolidate this section, or we can just leave it out in favor of linking to the Arrow documentation. 

Perhaps something like this:

```
The `Arrow IPC format <https://arrow.apache.org/docs/java/ipc.html>`_ defines two types of binary formats for serializing Arrow data: the streaming format and the file format (or random access format). Such files can be directly memory-mapped when read.
```

Reply:
Changed